deployments/puppet/manifests/init.pp

```diff
@@ -1,15 +1,52 @@
 # Main class that installs and configures the agent
 class signalfx_agent (
-    $config,
-    $package_stage = 'final',
-    $repo_base = 'dl.signalfx.com',
-    $config_file_path = '/etc/signalfx/agent.yaml',
-    $version = 'latest') {
+  $config,
+  $package_stage    = 'final',
+  $repo_base        = 'dl.signalfx.com',
+  $config_file_path = $::osfamily ? {
+    'debian'  => '/etc/signalfx/agent.yaml',
+    'redhat'  => '/etc/signalfx/agent.yaml',
+    'windows' => 'C:\\ProgramData\\SignalFxAgent\\agent.yaml',
+    'default' => '/etc/signalfx/agent.yaml'
+  },
+  $agent_version    = '',
+  $package_version = 'latest',
+) {
+
+  $service_name = 'signalfx-agent'
 
   if !$config['signalFxAccessToken'] {
     fail("The \$config parameter must contain a 'signalFxAccessToken'")
   }
 
+  if $::osfamily == 'windows' {
+    $agent_location = "C:\\Program Files\\SignalFx\\"
```

Comment:
This could be changed by a config option (i.e. if somebody were trying to install it to a different drive), so it shouldn't be hardcoded here.

Reply:
Added `installation_directory` config option